src/Compilers/VisualBasic/Portable/Binding/Binder.vb

```diff
@@ -452,6 +453,60 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return type.GetUseSiteErrorInfo()
         End Function
 
+        Private Function GetInternalXmlHelperType(syntax As VisualBasicSyntaxNode, diagBag As DiagnosticBag) As NamedTypeSymbol
+            Debug.Assert(Me.ContainingType IsNot Nothing)
+            Dim containingAssembly = Me.ContainingType.ContainingAssembly
+
+            Dim typeSymbol = GetInternalXmlHelperType(containingAssembly)
+
+            Dim useSiteError = GetUseSiteErrorForWellKnownType(typeSymbol)
+            If useSiteError IsNot Nothing Then
+                ReportDiagnostic(diagBag, syntax, useSiteError)
+            End If
+
+            Return typeSymbol
+        End Function
+
+        Private Shared Function GetInternalXmlHelperType(containingAssembly As AssemblySymbol) As NamedTypeSymbol
+            Const metadataName = "My.InternalXmlHelper"
+
+            Dim typeSymbol = containingAssembly.GetTypeByMetadataName(
```

Comment:
I can't tell from quick code inspection whether this takes into account root namespaces, but it would be good to at least have a simple testcase where My.InternalXmlHelper is nested in a root namespace.